api/announcements.py

```diff
@@ -32,17 +45,19 @@ def __init__(self, announcements):
         :return: A list of Announcement objects. The list will be empty if
             there are validation errors in `announcements`.
         """
+        self.problem = None
         validator = AnnouncementListValidator()
         validated = validator.validate_announcements(announcements)
         if isinstance(validated, ProblemDetail):
             # There's a problem with the way the announcements were
             # serialized to the database. Treat this as an empty list.
+            self.problem = validated
             validated = []
 
         self.announcements = [Announcement(**data) for data in validated]
 
     @property
-    def active(self):
+    def active(self) -> Generator["Announcement", None, None]:
```

Comment:
You should be able to use the `Announcements` class name (rather than a string) here by importing the `annotations` future at the top of the file.